pkg/scheduler/core/generic_scheduler.go

```diff
@@ -501,6 +502,17 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 				return
 			}
 			if fits {
+				// Iterate each plugin to verify current node
+				status := g.framework.RunFilterPlugins(pluginContext, pod, nodeName)
+				if !status.IsSuccess() {
+					predicateResultLock.Lock()
+					failedPredicateMap[nodeName] = append(failedPredicateMap[nodeName],
+						predicates.NewFailureReason(status.Message()))
+					errs[status.Message()]++
```

Comment:
I don't think this is correct. RunFilterPlugins should either return success, which means the node fits, or unschedulable, which means the node does not fit. Unschedulable is not an error.

Reply:
Correct. We should only do this when status code is `Error`.

Reply:
It has been addressed in the new diff

Reply:
@bsalamat shouldn't the condition be != unschedulable rather than == Error? The only two statuses that make sense for a filter to return is Success and Unschedulable, otherwise we should consider the plugin to have returned an internal error.